apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { Button, Modal } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { PageTemplatesPlugin } from '../index';
+import TemplateSelectorItem from './template-selector-item';
+import replacePlaceholders from '../utils/replace-placeholders';
+/* eslint-enable import/no-extraneous-dependencies */
+
+class SidebarModalOpener extends Component {
+	state = {
+		isTemplateModalOpen: false,
+		isWarningOpen: false,
+	};
+
+	toggleTemplateModal = () => {
+		this.setState( { isTemplateModalOpen: ! this.state.isTemplateModalOpen } );
+	};
+
+	toggleWarningModal = () => {
+		this.setState( { isWarningOpen: ! this.state.isWarningOpen } );
+	};
+
+	getLastTemplateUsed = ( { templateUsedSlug, templates } = this.props ) => {
```

Comment:
Any specific reason for adding this as a default parameter instead of just unpacking the props in the function body? Since it's a class method I didn't expect to see its instance variable be passed as a parameter.
Nitpick: Does `getLastUsedTemplate` sound better, or even `getCurrentTemplate` even though it's not completely accurate?

Reply:
> Any specific reason for adding this as a default parameter instead of just unpacking the props in the function body? 

Nope, just me being weird.  I didn't destructure anything at first and just had `this.props.propName` over and over again in that function.  Noah mentioned destructuring would make it look nicer and for some reason my first thought was to do it inside the function param.  LOL, on second look its just seems more readable to do it in the first line of the function body.

>Nitpick: Does getLastUsedTemplate sound better, or even getCurrentTemplate even though it's not completely accurate?

I updated it to `lastTemplateUsedSlug`, the `get` prefix to me seems to imply that it is a function that needs to be called when in this context it is just a string that has already been 'got'. =)